### ISSUE TITLE
rounded ui nodes cut off on the right and bottom sides

### ISSUE THREAD

**Issue by ickshonpe** [bevyengine/bevy#14955]
## Bevy version

main

## What went wrong

From the `borders` example:

<img width="500" alt="bad_border_radius" src="https://github.com/user-attachments/assets/5ab80a98-a796-48a0-b0ca-5b51ad77bbab">

**Comment by rparrett** [comment]
Confused about the issue after the title change. Is this about the yellow nodes or the red ones?

I can't reproduce the issue with the yellow nodes on macOS (m1) with `scale_factor` `1.0` or `2.0` on latest main.

I suspect that the red node behavior is not a regression, but I am unable to bisect due to whatever drama is happening at wgpu right now.

**Comment by oscrim** [comment]
Not sure if its related to this but I also encountered a border bug. When I run the example targeting x86 it all renders as it should but when the target is wasm32 its rendered incorrectly.
![image](https://github.com/user-attachments/assets/4623b97e-9f6d-4d83-8ff0-ae25405c80e8)

**Comment by oscrim** [comment]
> That's odd, the sdf should work the same on all platforms, it's just arithmetic.

I agree, I just tested it on my computer at work where the borders also render correctly. 

```
AdapterInfo { name: "ANGLE (Intel, Mesa Intel(R) UHD Graphics 770 (ADL-S GT1), OpenGL 4.6)", vendor: 32902, device: 0, device_type: IntegratedGpu, driver: "WebGL 2.0 (OpenGL ES", driver_info: "3.0 Chromium)", backend: Gl }
```

One thing that stands out when comparing the adapter info is that on the computers where its rendered correctly its using OpenGL 4.6 while the other computer uses OpenGL 4.5.0
